packages/admin-ui/client/pages/List/SortSelect.js

```diff
@@ -1,12 +1,75 @@
 /** @jsx jsx */
 import { jsx } from '@emotion/core';
-import { Component } from 'react';
+import { useMemo, useRef } from 'react';
 import styled from '@emotion/styled';
 import { ChevronDownIcon, ChevronUpIcon } from '@arch-ui/icons';
 import { OptionPrimitive, Options } from '@arch-ui/options';
 import { colors } from '@arch-ui/theme';
+import { Kbd } from '@arch-ui/typography';
 
-import { POPOUT_GUTTER } from '../../components/Popout';
+import { DisclosureArrow, Popout, POPOUT_GUTTER } from '../../components/Popout';
+import { useList, useListSort, useKeyDown } from './dataHooks';
+
+type Props = {
+  listKey: string,
+};
+
+export default function SortPopout({ listKey }: Props) {
+  const list = useList(listKey);
+  const [sortValue, handleSortChange] = useListSort(listKey);
+  const altIsDown = useKeyDown('Alt');
+  const popoutRef = useRef();
+
+  const handleChange = field => {
+    const isSelected = field.path === sortValue.field.path;
+    console.log('handleChange', field);
+
+    let direction = 'ASC';
+    if (isSelected) {
+      direction = invertDirection(sortValue.direction);
+    } else if (altIsDown) {
+      direction = 'DESC';
+    }
+
+    handleSortChange({ field, direction });
+    popoutRef.current.close();
+  };
+
+  const cachedOptions = useMemo(() => list.fields.map(({ options, ...field }) => field));
```

Comment:
```suggestion
  const cachedOptions = useMemo(() => list.fields.map(({ options, ...field }) => field), [list]);
```

Reply:
Not sure on the nuances of deps here, but is using an object the right way to go? What about `list.key`?

Reply:
Using an object is completely fine. Using `list.key` would technically work in this exact case but you should use whatever values are used in the function. `list.fields` would also work but since we know that `list.fields` will only change when `list` changes, using `list` is probably better since it's shorter.